File: Year 1/Semester 2/Java/Lessons/classes/university.java
package  classes;

public class university {

    public String name;
    private int year;

    // constructor method

    public university(String name, int year) {
        // this is used
        this.name = name;
        this.year = year;
        graduate();

    }

    public void speak() {
        System.out.println("I am in the " + this.name + "and I am in year " + this.year);
    }  

    public int getYear() {
        return this.year();
    }

    // change the university or year?

    public void setYear(int year) {
        this.year = year;
    }

    // private method
    private int graduate() {
        return this.year + 2;
    }
}

// object will be created in Main - file discussion

File: Year 1/Semester 1/C/Lessons/classes/university.c
#include <stdio.h>
#include <string.h>

// University struct to represent the class
typedef struct {
    char name[50];
    int year;
} University;

// Constructor function for University
University createUniversity(const char *name, int year) {
    University u;
    strcpy(u.name, name);
    u.year = year;
    u.year = graduate(u);  // Call the private method equivalent
    return u;
}

// Method to display university details
void speak(University u) {
    printf("I am in the %s and I am in year %d\n", u.name, u.year);
}

// Getter for year
int getYear(University u) {
    return u.year;
}

// Setter for year
void setYear(University *u, int year) {
    u->year = year;
}

// Private method equivalent to 'graduate'
static int graduate(University u) {
    return u.year + 2;
}

int main() {
    // Create an object in main
    University micheal = createUniversity("tuks", 1900);
    speak(micheal);

    // Call setter
    setYear(&micheal, 2400);
    speak(micheal);

    // Access year via getter
    int year = getYear(micheal);
    printf("The current year is %d\n", year);

    return 0;
}

